sandstone/lib/tests/python/test_filewatcher.py

```diff
@@ -0,0 +1,66 @@
+import unittest
+import mock
+from sandstone.filewatcher import Filewatcher
+import tempfile
+import os
+from sandstone.apps.filebrowser.posixfs import PosixFS
+import time
+import copy
+
+
+class FilewatcherTestCase(unittest.TestCase):
+    def setUp(self):
+        self.test_dir = tempfile.mkdtemp()
+
+    def tearDown(self):
+        # remove all watches
+        dirs = copy.deepcopy(Filewatcher._watches)
+        for key, value in dirs.items():
+            Filewatcher.remove_directory_to_watch(key)
+
+    def test_add_directory_to_watch(self):
+        self.assertEqual(len(Filewatcher._watches), 0)
+        Filewatcher.add_directory_to_watch(self.test_dir)
+        self.assertEqual(len(Filewatcher._watches), 1)
+
+    def test_remove_directory_to_watch(self):
```

Comment:
Cascading tests like this should be avoided when possible.

Also, given that `tearDown()` removes all watches, and is called after each test method, I'm not sure this is exercising the intended path.